react setup-ssr doesn't take rspack into account

### Current Behavior

Due to #28103 I'm stuck with Nx 18.3.5 and I have no idea if this is relevant for the newest version.

The conf created by `@nx/react:setup-ssr` doesn't take rspack into account and looks for a `webpack.config.js` file

### Expected Behavior

`setup-ssr` finds out we are using rspack and creates a working configuration

### GitHub Repo

_No response_

### Steps to Reproduce

1. create a React app, use rspack
2. run the `setup-ssr` generator
3. try to serve the app


### Nx Report

```shell
NX   Report complete - copy this into the issue template

Node   : 16.20.2
OS     : darwin-arm64
yarn   : 3.8.5

nx                 : 18.3.5
@nx/js             : 19.8.0
@nrwl/js           : 18.3.5
@nx/jest           : 18.3.5
@nx/linter         : 18.3.5
@nx/eslint         : 18.3.5
@nx/workspace      : 18.3.5
@nx/cypress        : 18.3.5
@nx/devkit         : 18.3.5
@nx/eslint-plugin  : 18.3.5
@nx/express        : 18.3.5
@nx/node           : 18.3.5
@nx/react          : 18.3.5
@nrwl/tao          : 18.3.5
@nx/web            : 18.3.5
@nx/webpack        : 18.3.5
typescript         : 5.4.5
---------------------------------------
Registered Plugins:
@nx/eslint/plugin
@nx/cypress/plugin
@nx/jest/plugin
@nx/webpack/plugin
---------------------------------------
Community plugins:
@nx/rspack : 19.7.1
---------------------------------------
The following packages should match the installed version of nx
  - @nx/js@19.8.0

To fix this, run `nx migrate nx@19.8.0`
```


### Failure Logs

```shell
❯ yarn nx serve ssr-client

> nx run ssr-client:serve:development

node:internal/modules/cjs/loader:1028
  const err = new Error(message);
              ^

Error: Cannot find module '/redacted/packages/ssr-client/webpack.config.js'
Require stack:
```


### Package Manager Version

_No response_

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

_No response_

--- a/packages/react/src/generators/setup-ssr/setup-ssr.ts
+++ b/packages/react/src/generators/setup-ssr/setup-ssr.ts
@@ -124,6 +124,12 @@ export async function setupSsrGenerator(tree: Tree, options: Schema) {
     );
   }
 
+  if (projectConfig.targets.build.executor === '@nx/rspack:rspack') {
+    options.bundler = 'rspack';
+  } else if (projectConfig.targets.build.executor === '@nx/webpack:webpack') {
+    options.bundler = 'webpack';
+  }
+
   projectConfig.targets = {
     ...projectConfig.targets,
     server: {

